src/prefect/tasks.py

```diff
@@ -466,6 +466,10 @@ def __init__(
             ):
                 persist_result = True
 
+        # Check for global cache disable setting
+        if settings.tasks.disable_cache:
+            persist_result = False
```

Comment:
I think this could lead to surprising edge cases; was there a reason you chose to do this vs. a more targeted

```python
cache_policy = NO_CACHE
```

?